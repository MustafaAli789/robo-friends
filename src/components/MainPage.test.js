import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage'

let wrapper;
beforeEach(()=>{
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false
    }
    wrapper = shallow(<MainPage {...mockProps} />)
})

it('rendes main page without crashin', () => {
    expect(wrapper.debug()).toMatchSnapshot()
});

it('filters robotos correctly', () => {
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: '3',
            name: 'john',
            email: 'john@gmail.com'
        }],
        searchField: 'john',
        isPending: false
    }
    const wrapper2 = shallow(<MainPage {...mockProps2} />)
    expect(wrapper2.instance().filterRobots()).toEqual([{
        id: '3',
        name: 'john',
        email: 'john@gmail.com'
    }]);
});

it('filters robotos correctly2', () => {
    const mockProps3 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: '3',
            name: 'john',
            email: 'john@gmail.com'
        }],
        searchField: 'q',
        isPending: false
    }
    const filteredRobots = []
    const wrapper3 = shallow(<MainPage {...mockProp32} />)
    expect(wrapper3.instance().filterRobots()).toEqual(filteredRobots);
});