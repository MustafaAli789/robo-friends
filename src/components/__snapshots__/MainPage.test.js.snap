// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendes main page without crashin 1`] = `
"<div className=\\"tc\\">
  <Header />
  <SearchBox searchChange={[undefined]} />
  <Scroll>
    <ErrorBoundry>
      <CardList robots={{...}} />
    </ErrorBoundry>
  </Scroll>
</div>"
`;
